src/releng_docs/services_shipit.rst

```diff
@@ -57,3 +60,44 @@ It does the following tasks on every run:
 
 
 .. _libmozdata: https://github.com/mozilla/libmozdata/
+
+
+``src/shipit_pipeline``
+
+TODO
+
+
+Steps
+-----
+
+Step Services API
+~~~~~~~~~~~~~~~~~
+
+In order to ensure the Pipeline service can successfully managed Steps, each Step Service is required to implement the following API:
+(TODO, flesh this out more)
+
+* /
+** GET - Returns all steps with status (TODO: probably need pagination and filtering here)
+* /{uid}
+** PUT - Create a new Step
+** DELETE - Remove the given Step
+* /{uid}/definition
+** GET Returns the definition of the given step
+* /{uid}/status
+** GET Returns the status of the given step.
+*** Currently, one of: starting, running, stopping, exception, completed, failed
```

Comment:
I'd like to figure out the possible statuses. I'm not sure what we really gain from having [starting, running, stopping] vs just [running]. Particularly if we have a way to retrieve custom  service specific data summarizing state

As per last week's meeting, I was thinking we could try a minimal set of 4 statuses similar to aws step functions: in_progress, success, failure, cancelled